test/models/parcel_item_test.rb

```diff
@@ -54,5 +54,20 @@
 
 class ParcelItemTest < ActiveSupport::TestCase
   test_model_actions
-  # Add tests here...
+
+  test "give doesn't create the dependent records if there is an exception during the process" do
+    product = create(:product)
+    parcel_item = create(:parcel_item, product: product, product_identification_number: '12345678', product_name: 'Product name')
+    ProductMovement.destroy_all
+    ProductLocalization.destroy_all
+    ProductEnjoyment.stub :create!, lambda { |*args| raise } do
+      begin
+        parcel_item.give
+      rescue
+        puts "Error message"
```

Comment:
Can you remove this `puts` that will not make much sens when looking through the test logs ?